### PR TITLE
fix: Upload remove margin using grid display

### LINKED ISSUE #47992: Upload picture-card 可以更改为 flex   gap 布局，不会有多余的右 margin 和 下 margin
### Reproduction link

[https://ant.design/components/upload-cn#components-upload-demo-picture-card](https://ant.design/components/upload-cn#components-upload-demo-picture-card)

### Steps to reproduce

查看最右侧和最下侧的 margin

### What is expected?

最右侧和最下侧的没有 margin

### What is actually happening?

最右侧和最下侧的有 margin

| Environment | Info |
| --- | --- |
| antd | 5.15.3 |
| React | latest |
| System | mac |
| Browser | chrome |

---
但是也要考虑下是不是会和 Tag 的 margin 一样会有 break change

### PR DESCRIPTION
**Pull request by CooperHash**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [X] Other (about what?)

### 🔗 Related issue link

- close #47992

### 💡 Background and solution

改动前

image

改动后

image

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    --       |
| 🇨🇳 Chinese |    --       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

File: components/upload/style/picture.ts [hunk 1]
```diff
@@ -138,19 +136,26 @@ const genPictureCardStyle: GenerateStyle<UploadToken> = (token) => {
 
       // list
       [`${listCls}${listCls}-picture-card, ${listCls}${listCls}-picture-circle`]: {
+        display: 'grid',
```
Comment (li-jia-nan) [inline components/upload/style/picture.ts:139]: 为啥不用 flex ？

Comment (CooperHash) [reply components/upload/style/picture.ts:139]: 没做太多思考，这里用flex会更好吗

Comment (li-jia-nan) [reply components/upload/style/picture.ts:139]: 我是看到 #47992 这个 issue 中讨论的是 flex，grid 布局在 antd 里面用的比较少

File: components/upload/style/picture.ts [hunk 2]
```diff
@@ -138,19 +136,25 @@ const genPictureCardStyle: GenerateStyle<UploadToken> = (token) => {
 
       // list
       [`${listCls}${listCls}-picture-card, ${listCls}${listCls}-picture-circle`]: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: `${token.marginXS} ${token.marginXS}`,
```
Comment (li-jia-nan) [inline components/upload/style/picture.ts:141]:
```suggestion
        gap: token.marginXS,
```

**Comment by li-jia-nan** [comment]
感觉这个得放在 feature 分支，我不确定，大佬们啥意见？@zombieJ @MadCcc

**Comment by afc163** [comment]
+1 feature 分支

**Comment by CooperHash** [comment]
那我是关掉这个分支，再重写提交一次到feature吗

**Comment by li-jia-nan** [comment]
可以直接改，上面有编辑按钮

File: components/upload/style/picture.ts [hunk 3]
```diff
@@ -138,19 +136,25 @@ const genPictureCardStyle: GenerateStyle<UploadToken> = (token) => {
 
       // list
       [`${listCls}${listCls}-picture-card, ${listCls}${listCls}-picture-circle`]: {
+        display: 'flex',
+        flexWrap: 'wrap',
+        gap: token.marginXS,
```
Comment (MadCcc) [inline components/upload/style/picture.ts:141]: gap 在 chromium 84 版本以上才支持，需要用 @supports 做下降级

**Comment by CooperHash** [comment]
git被我搞炸了...我重新开一个branch吧...

**Comment by CooperHash** [comment]
#48091

**Pull request closed without merging**